### PR TITLE
Update vecgeom to 1.1.16

### LINKED ISSUE #6974: Migrate VecGeom package to the new version 1.1.16
In view of preparation for 12_0_X for Run-3 a new version of VecGeom package needs to be installed: https://gitlab.cern.ch/VecGeom/VecGeom/-/tree/v1.1.16 

According to VecGeom developer comments we may expect regression, when switch to this bug fix version.

### PR DESCRIPTION
**Pull request by mrodozov**
Resolves: #6974 
`mic mic_avx512` are not in the original CMakeLists anymore

### CODE REVIEW

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
please test for slc7_aarch64_gcc9

File: vecgeom.spec [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+### RPM external vecgeom v1.1.16
 ## INCLUDE compilation_flags
-%define tag 5a275d77ef80b12240d59fd276231ad50d5df577
+%define tag ed9a40412c354652262ec80af449f5531206e52c
 Source: git+https://gitlab.cern.ch/VecGeom/VecGeom.git?obj=master/%{tag}&export=%{n}-%{realversion}&output=/%{n}-%{realversion}.tgz
 BuildRequires: cmake gmake
 %define keep_archives true
 
-Patch0: vecgeom-fix-for-arm64
+#Patch0: vecgeom-fix-for-arm64
```
Comment (smuzaffar) [inline vecgeom.spec]: please remove these lines if patch is not needed

File: vecgeom.spec [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
 Patch1: vecgeom-fix-vector
 
 %prep
 %setup -n %{n}-%{realversion}
 
-%patch0 -p1
+#%patch0 -p1
```
Comment (smuzaffar) [inline vecgeom.spec]: this too

File: vecgeom.spec (hunk 1 above)
Comment (mrodozov) [reply vecgeom.spec]: sorry i missed that

**Comment by mrodozov** [comment]
abort

**Comment by mrodozov** [comment]
please test

File: vecgeom.spec (hunk 2 above)
Comment (mrodozov) [reply vecgeom.spec]: test remainings :/

**Comment by mrodozov** [comment]
please test for slc7_aarch64_gcc9

**Comment by mrodozov** [comment]
please test for slc7_ppc64le_gcc9

**Comment by mrodozov** [comment]
please test for CMSSW_12_0_X/slc7_ppc64le_gcc9

**Comment by civanch** [comment]
My impression that unit tests problems are not connected with this PR.

**Comment by mrodozov** [comment]
they are not

**Comment by mrodozov** [comment]
+externals

**Pull request merged**